bundled/muparser_v2_2_4/include/muParserCallback.h

```diff
@@ -81,6 +81,9 @@ class ParserCallback
     ParserCallback(strfun_type3 a_pFun, bool a_bAllowOpti);
     ParserCallback();
     ParserCallback(const ParserCallback &a_Fun);
+
+    // deall.II: added the following line to remove Wdeprecated-copy warnings:
+    constexpr ParserCallback& operator=(const ParserCallback&)=default;

```

Comment:
What does `constexpr` here imply? That the returned object is a `const` reference or that the function can be called in a `constexpr` context? I don't think that the former can be right because what the function returns is the current object (which isn't `const`).

Reply:
`constexpr` functions are only `const` for C++11, not for later standards. It implies that the `function` can be called in a `constexpr` context.
Since there is no `constexpr` constructor, there is also no reason to make the copy constructor `constexpr` in my opinion.

Reply:
gcc complains when I remove the ``constexpr``, by the way.